lib/src/features/chat/data/models/contact_model.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:doormer/src/features/chat/domain/entities/contact_entity.dart';
+import 'package:uuid/uuid.dart';
+
+class ContactModel {
+  final String id; // UUID as a string
+  final String userName;
+  final String avatarUrl;
+  final String lastMessage;
+  final DateTime createdTime;
+  final bool isArchived;
+  final bool isRead;
+
+  ContactModel({
+    required this.id,
+    required this.userName,
+    required this.avatarUrl,
+    required this.lastMessage,
+    required this.createdTime,
+    required this.isArchived,
+    required this.isRead,
+  });
+
+  /// Factory constructor to create a [ContactModel] from JSON.
+  factory ContactModel.fromJson(Map<String, dynamic> json) {
+    if (!Uuid.isValidUUID(fromString: json['id'])) {
```

Comment:
I removed it because I thought the backend guarantees that the response will have all required fields are non-nulls and non-empty. So we only need to ensure it's a valid uuid format. I can add it back if needed.